openedx/core/lib/request_utils.py

```diff
@@ -313,36 +311,35 @@ def _log_and_monitor_expected_errors(request, exception, caller):
         cached_module_and_class = cached_handled_exception.value
         # exception was already processed by a different caller
         if cached_handled_exception.value == module_and_class:
-            set_custom_attribute('checked_error_expected_from', 'multiple')
+            set_custom_attribute('checked_error_ignored_from', 'multiple')
             return
 
         # We have confirmed using monitoring that it is very rare that middleware and drf handle different uncaught exceptions.
-        # We will leave this attribute in place, but it is not worth investing in a workaround, especially given that
-        # New Relic now offers its own expected error functionality, and this functionality may be simplified or removed.
+        # We will leave this attribute in place, but it is not worth investing in a workaround.
+        # New Relic now offers its own ignored error functionality, and this functionality may be simplified or removed.
         set_custom_attribute('unexpected_multiple_exceptions', cached_module_and_class)
         log.warning(
-            "Unexpected scenario where different exceptions are handled by _log_and_monitor_expected_errors. "
+            "Unexpected scenario where different exceptions are handled by _log_and_monitor_ignored_errors. "
             "See 'unexpected_multiple_exceptions' custom attribute. Skipping exception for %s.",
             module_and_class,
         )
         return
     request_cache.set('handled_exception', module_and_class)
 
-    if module_and_class not in expected_error_settings_dict:
+    if module_and_class not in ignored_error_settings_dict:
         return
 
     exception_message = str(exception)
-    set_custom_attribute('error_expected', True)
 
-    expected_error_settings = expected_error_settings_dict[module_and_class]
-    if expected_error_settings['is_ignored']:
+    ignored_error_settings = ignored_error_settings_dict[module_and_class]
+    if ignored_error_settings['is_ignored']:
```

Comment:
This if conditional will always be True now. We need the lines after the if statement, but not the if statement itself.

Reply:
@robrap but here we are removing is_ignored so what should we do in this case?

Reply:
This is what the new code would look like:
```
    # Additional error details are needed for ignored errors, because they are otherwise
    # not available by our monitoring system, because they have been ignored.
    set_custom_attribute('error_ignored_class', module_and_class)
    set_custom_attribute('error_ignored_message', exception_message)

    ignored_error_settings = ignored_error_settings_dict[module_and_class]
    if ignored_error_settings['log_error']:
        ...
```
What I changed:
1. I took the code from inside the if, and simply outdented it (the opposite of indenting).
2. I moved the `ignored_error_settings = ...` a little lower, because it isn't getting used until the next if statement.

Reply:
@robrap yes that makes sense. is_ignored this will removed.